src/runtime/virtcontainers/kata_agent.go

```diff
@@ -394,13 +394,27 @@ func setupSandboxBindMounts(sandbox *Sandbox) error {
 	if err := os.MkdirAll(sandboxMountDir, DirMode); err != nil {
 		return fmt.Errorf("Creating sandbox shared mount directory: %v: %w", sandboxMountDir, err)
 	}
+	var mountedList []string
+	defer func() {
+		if err != nil {
+			for _, mnt := range mountedList {
+				if err := syscall.Unmount(mnt, syscall.MNT_DETACH|UmountNoFollow); err != nil {
```

Comment:
Here you're overriding `err`, is that what you want to do?  I'd go for an `err2` or similar, as I think the error we want to return is not the one from the unmount, but rather the one coming from the BindMount.

Reply:
I *think* / *thought* it would be scoped to the if block here? Let me test...

Reply:
It actually is.